### PR TITLE
feat: App support component

### LINKED ISSUE #44716: [功能请求]: `<App />` 组件添加选项控制是否需要 DOM 元素和 .ant-app 样式重置
### What problem does this feature solve?

仅使用 `<App />` 为 message / modal / notification 提供 context 时，会附带不需要的 DOM 和样式重置

### What does the proposed API look like?

```tsx 
<App abstract={boolean}>
  {...}
</App>
```

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #44716

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   App support `component` for customization.        |
| 🇨🇳 Chinese |  App 支持 `component` 以自定义渲染元素。        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at c2c84ab</samp>

Added a new prop `component` to the `App` component to allow customizing or removing the root element. Updated the documentation and tests for the `App` component in both English and Chinese. Fixed the import order of some files to follow ESLint rule.

### <samp>🤖 Generated by Copilot at c2c84ab</samp>

*  Add a new `component` prop to the `App` component to allow customizing or disabling the render element ([link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bR20), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bR34), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bL63-R81))
*  Update the documentation of the `App` component in English and Chinese to include the `component` prop and its usage examples ([link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-c779f7d3b6a66acd0ce5e8bf4f1625badb983056711cc82435808368588ed38cR133), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-88905ff2c3941f956e0d1dbf8871b8f2d633992532d79997fc5be05032729291R134), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-c779f7d3b6a66acd0ce5e8bf4f1625badb983056711cc82435808368588ed38cL32-R33), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-c779f7d3b6a66acd0ce5e8bf4f1625badb983056711cc82435808368588ed38cL105-R107))
*  Add two test cases to verify the behavior of the `component` prop in the `App` component ([link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8R212-R233))
*  Adjust the import order of external and internal modules in the source code and documentation of the `App` component to follow the ESLint rule ([link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8L1-R4), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-c779f7d3b6a66acd0ce5e8bf4f1625badb983056711cc82435808368588ed38cL32-R33), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-c779f7d3b6a66acd0ce5e8bf4f1625badb983056711cc82435808368588ed38cL105-R107), [link](https://github.com/ant-design/ant-design/pull/45292/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bL1-R4))

### CODE REVIEW

File: components/app/index.en-US.md [hunk 1]
```diff
@@ -129,6 +130,7 @@ Common props ref：[Common props](/docs/react/common-props)
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
+| component | Config render element, if `false` will not create DOM node | ComponentType | div | 5.11.0 |
```
Comment (kiner-tang) [inline components/app/index.en-US.md:133]: 命名上是不是再讨论一下，感觉用 component 怪怪的

Comment (kiner-tang) [reply components/app/index.en-US.md:133]: 这个如何：`renderElement: () => ComponentType`

Comment (crazyair) [reply components/app/index.en-US.md:133]: 这里沿用的 Form 的 component

**Review by MadCcc — APPROVED** [review]

File: components/app/index.en-US.md (hunk 1 above)
Comment (MadCcc) [reply components/app/index.en-US.md:133]: component 在 antd 的命名规范里

Comment (Wxh16144) [reply components/app/index.en-US.md:133]: 应该不用，因为 Table 也有一个 components props 功能是一样的，统一命名

Comment (zombieJ) [reply components/app/index.en-US.md:133]: `component` 是标准命名，Flex、Form 都是这个~

File: components/app/index.tsx [hunk 2]
```diff
@@ -60,15 +64,22 @@ const App: React.FC<AppProps> & { useApp: typeof useApp } = (props) => {
     [messageApi, notificationApi, ModalApi],
   );
 
+  // ============================ Render ============================
+  const Component = component === false ? React.Fragment : component;
```
Comment (crazyair) [inline components/app/index.tsx:68]: 类型抽离了一个 CustomComponent，有没有想过这里用一块抽离，抽离一个 hooks 或者 utils 达到各个组件支持的 component 逻辑一致性

Comment (Wxh16144) [reply components/app/index.tsx:68]: 没法复用吧

Comment (crazyair) [reply components/app/index.tsx:68]: 可以抽离组件
```tsx
<CustomComponent component={component}>
     xxx
</CustomComponent>
```

Comment (zombieJ) [reply components/app/index.tsx:68]: 嗯。因为 Form 的是在 rc-field-form 里处理，Flex 是不支持 `false`。大家都有一点差异，暂时还不用~

File: components/app/index.en-US.md (hunk 1 above)
Comment (kiner-tang) [reply components/app/index.en-US.md:133]: 搜嘎

**Review by kiner-tang — APPROVED** [review]

File: components/app/index.tsx (hunk 2 above)
Comment (crazyair) [reply components/app/index.tsx:68]: Flex 可以支持 false，不就是 <></> 嘛。

**Pull request merged**